In [1]:
from math import factorial

## Задача 1
### Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Как уточнили на уроке, порядок нам важен. Поэтому используем размещения.

In [2]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

In [3]:
arrangements(10, 4)

5040

Мы получили 5040 способов

## Задача 2
### Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Здесь необходимо использовать сочетания, т.к. порядок нам не важен.

Формула:

In [5]:
def combinations(n, k):
    return factorial(n) // (factorial(k) * factorial(n - k))

Сначала рассчитаем, когда вытянут 1 туз из 4 карт, а потом 3 оставшиеся из 52. По аналогии считаем когда вытянуто 2 туза, потом 3, потом 4. 

In [6]:
a = combinations(4, 1) * combinations(52, 3)
b = combinations(4, 2) * combinations(52, 2)
c = combinations(4, 3) * combinations(52, 1)
d = combinations(4, 4) * combinations(52, 0)

In [7]:
summ = a+b+c+d
print(summ)

96565


Получаем, что кол-во исходов равно 96565

## Задача 3
### Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

Всего пар на скамейке может быть 6. Пара может сесть либо AB либо BA, т.е. двумя способами, т.е. 2!. Все остальные получается любыми оставшимися 5 способоми, соответственно это равно 5!. Всего мест 7, значит количество исходов 7!

In [10]:
probability = (6 * factorial(2) * factorial(5)) / factorial(7)
print(probability)

0.2857142857142857


## Задача 4
### Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Сначала нужно рассчитать вероятность вытянуть все три билета, которые студент знает. Т.к. порядок не важен, то это будет сочетания.

In [ ]:
Рассчитаем, что мы выбрали 3 из 50 и 3 из 60. И разделим одно на другое

In [13]:
a = combinations(50, 3)
b = combinations(60, 3)
c = a/b
print(c)

0.5727644652250146


Далее нужно взять 2 билета из 50, т.к. мы их знаем и 1 билет из тех, которые не знаем - из 10. И поделить на все вытянутые из 60

In [19]:
a1 = combinations(50, 2) * combinations(10, 1)
b1 = combinations(60, 3)
c1 = a1/b1
print(c1)

0.35797779076563413


## Задача 5
### Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

Проверим, являются ли события независимыми по формуле P(B|A) = P(B)

Найдем вероятность появления четного числа P(A) = 3/6 = 1/2

Найдем вероятность появления числа больше при условии появления четного числа, P(B|A). Получается четных после трех - 2. Получаем 2/6

Т.к. по формуле P(B|A) = P(B), а у нас 2/6 не равно 1/2, соответственно, события являются зависимыми.

## Задача 6 (Дополнительно)
### Допустим, имеется некоторая очень редкая болезнь (поражает 0.1 % населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99 % больных этой болезнью и всего лишь в 1 % случаев даёт ложный положительный ответ.
### Вопрос: какова вероятность, что вы действительно больны ей?
### Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.